src/content/actions/receive-profile.js

```diff
@@ -232,24 +234,91 @@ export function receiveProfileFromWeb(profile: Profile): ThunkAction {
   };
 }
 
-export function errorReceivingProfileFromWeb(error: any): Action {
+export function temporaryErrorReceivingProfileFromWeb(error: TemporaryError): Action {
   return {
-    type: 'ERROR_RECEIVING_PROFILE_FROM_WEB',
+    type: 'TEMPORARY_ERROR_RECEIVING_PROFILE_FROM_WEB',
     error,
   };
 }
 
+export function fatalErrorReceivingProfileFromWeb(error: Error): Action {
+  return {
+    type: 'FATAL_ERROR_RECEIVING_PROFILE_FROM_WEB',
+    error,
+  };
+}
+
+function wait(delayMs) {
+  return new Promise(resolve => setTimeout(resolve, delayMs));
+}
+
+type FetchProfileArgs = {
+  url: string,
+  onTemporaryError: TemporaryError => void,
+};
+/**
+ * Tries to fetch a profile on URL url. If the profile is not found,
+ * onTemporaryError is called with an appropriate error, we wait 1 second, and
+ * then tries again. If we still can't find the profile after 11 tries, we throw
+ * a permanent error.
+ * @param {string} url URL to fetch
+ * @param {Function} onTemporaryError callback that will be called for
+ * tempoerary errors.
+ * @returns {Promise.<String>} Resolves to the text profile if successful,
+ * rejects with an error for fatal errors.
+ */
+async function _fetchProfile({ url, onTemporaryError }: FetchProfileArgs) {
+  const MAX_WAIT_SECONDS = 10;
+  let i = 0;
+
+  while (true) { // eslint-disable-line no-constant-condition
```

Comment:
Does eslint complain about `for (;;)`? That's maybe the more idiomatic way to do this (read as "forever").